### PR TITLE
[Bug](compaction) pass arena to function->add_batch_range

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes

*** Query id: 0-0 ***
*** tablet id: 41001 ***
*** Aborted at 1706668862 (unix time) try "date -d @1706668862" if you are using GNU date ***
*** Current BE git commitID: 37d31a5 ***
*** SIGSEGV address not mapped to object (@0x10) received by PID 3588 (TID 5429 OR 0x7f8e505cd700) from PID 16; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /root/src/doris-2.0/be/src/common/signal_handler.h:417
 1# os::Linux::chained_handler(int, siginfo*, void*) in /root/doris/java8/jre/lib/amd64/server/libjvm.so
 2# JVM_handle_linux_signal in /root/doris/java8/jre/lib/amd64/server/libjvm.so
 3# signalHandler(int, siginfo*, void*) in /root/doris/java8/jre/lib/amd64/server/libjvm.so
 4# 0x00007F90D46CB400 in /lib64/libc.so.6
 5# doris::vectorized::AggregateFunctionCollectSetData<doris::StringRef, std::integral_constant<bool, false> >::merge(doris::vectorized::AggregateFunctionCollectSetData<doris::StringRef, std::integral_constant<bool, false> > const&, doris::vectorized::Arena*) at /root/src/doris-2.0/be/src/vec/aggregate_functions/aggregate_function_collect.h:147
 6# doris::vectorized::IAggregateFunctionHelper<doris::vectorized::AggregateFunctionNullUnaryInline<doris::vectorized::AggregateFunctionCollect<doris::vectorized::AggregateFunctionCollectSetData<doris::StringRef, std::integral_constant<bool, false> >, std::integral_constant<bool, false> >, false> >::deserialize_and_merge_from_column_range(char*, doris::vectorized::IColumn const&, unsigned long, unsigned long, doris::vectorized::Arena*) const at /root/src/doris-2.0/be/src/vec/aggregate_functions/aggregate_function.h:462
 7# doris::vectorized::IAggregateFunctionHelper<doris::vectorized::AggregateStateUnion>::add_batch_range(unsigned long, unsigned long, char*, doris::vectorized::IColumn const**, doris::vectorized::Arena*, bool) in /root/doris/be/lib/doris_be
 8# doris::vectorized::VerticalBlockReader::_update_agg_value(std::vector<COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>, std::allocator<COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn> > >&, int, int, bool) at /root/src/doris-2.0/be/src/vec/olap/vertical_block_reader.cpp:302
 9# doris::vectorized::VerticalBlockReader::_update_agg_data(std::vector<COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>, std::allocator<COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn> > >&) at /root/src/doris-2.0/be/src/vec/olap/vertical_block_reader.cpp:284
10# doris::vectorized::VerticalBlockReader::_agg_key_next_block(doris::vectorized::Block*, bool*) in /root/doris/be/lib/doris_be

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by HappenLee** [comment]
We better add regression test to cover the case

File: be/src/vec/olap/vertical_block_reader.h [hunk 1]
```diff
@@ -107,6 +107,7 @@ class VerticalBlockReader final : public TabletReader {
     // for agg mode
     std::vector<AggregateFunctionPtr> _agg_functions;
     std::vector<AggregateDataPtr> _agg_places;
+    Arena _arena;
```
Comment (yiguolei) [inline be/src/vec/olap/vertical_block_reader.h:110]: Do we need clear the arena when one segment flushed? to reduce the memory??

**Comment by hello-stephen** [comment]
run performance

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/vec/olap/vertical_block_reader.cpp [hunk 2]
```diff
@@ -319,6 +319,9 @@ void VerticalBlockReader::_update_agg_value(MutableColumns& columns, int begin,
             function->reset(place);
         }
     }
+    if (is_close) {
+        _arena.clear();
```
Comment (HappenLee) [inline be/src/vec/olap/vertical_block_reader.cpp:323]: maybe cause access after freeed?

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by mrhhsg — APPROVED** [review]
LGTM

**Pull request merged**